File: 299.getHint.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
        int A = 0;
        int B = 0;
        unordered_map<char, int> sm;
        unordered_map<char, int> gm;
        
        for (int i = 0; i < secret.size(); i++) {
            if (secret[i] == guess[i]) A++;
            sm[secret[i]]++;
            gm[guess[i]]++;
        }

        for (int i = 0; i < 10; i++) {
            cout << to_string(i) << " " << sm['0'+i] << " " << gm['0'+i] << endl;
            B += min(sm['0'+i], gm['0'+i]);
        }

        return to_string(A)+'A'+to_string(B-A)+'B';
    }
};

File: cool/299.getHint.java
public String getHint(String secret, String guess) {
        int[] bucket = new int[10];
        int bull = 0;
        int cow = 0;
        for(int i=0;i<secret.length();i++){
            if(secret.charAt(i)== guess.charAt(i)){
                bull++;
                continue;
            }
            bucket[secret.charAt(i) - '0'] += 1;
            bucket[guess.charAt(i) - '0'] -= 1;

        }
        //计算bucket中正值的个数
        for(int i=0;i<10;i++){
            if(bucket[i] > 0)
                cow+= bucket[i];
        }
        
        cow = secret.length() - cow - bull;
        String res = bull + "A" + cow + "B";
        return res;
    }